WordPress/Classes/ViewRelated/Support/SupportTableViewController.swift

```diff
@@ -183,24 +188,66 @@ private extension SupportTableViewController {
     func contactUsSelected() -> ImmuTableAction {
         return { [unowned self] row in
             self.tableView.deselectSelectedRowWithAnimation(true)
-            if ZendeskUtils.zendeskEnabled {
-                guard let controllerToShowFrom = self.controllerToShowFrom() else {
-                    return
-                }
-                ZendeskUtils.sharedInstance.showNewRequestIfPossible(from: controllerToShowFrom, with: self.sourceTag) { identityUpdated in
-                    if identityUpdated {
-                        reloadViewModel()
-                    }
-                }
-            } else {
-                guard let url = Constants.forumsURL else {
-                    return
-                }
-                UIApplication.shared.open(url)
+            self.startContactUsFlow()
+        }
+    }
+
+    private func startContactUsFlow() {
+        // If Zendesk is not enabled, bail out and load the forums webpage instead.
+        guard ZendeskUtils.zendeskEnabled else {
+            loadForums()
+            return
+        }
+
+        guard let controllerToShowFrom = controllerToShowFrom() else {
+            return
+        }
+
+        let selfServiceFlowAvailable = true
+
+        if selfServiceFlowAvailable {
+            contactSupportSelfServiceController.present(from: controllerToShowFrom)
+        } else {
+            presentZendesk(from: controllerToShowFrom)
+        }
+    }
+
+    private func presentZendeskOrLoadForums() {
+        // If Zendesk is not enabled, bail out and load the forums webpage instead.
+        guard ZendeskUtils.zendeskEnabled, let presenter = controllerToShowFrom() else {
+            return loadForums()
+        }
+
+        if presenter.presentedViewController != nil {
+            // First dismiss whatever's on screen, then handover to Zendesk. This is to avoid
+            // failures due to the Zendesk SDK trying to present from a view controller that
+            // is already presenting something.
+            presenter.dismiss(animated: true) { [weak self] in
+                self?.presentZendesk(from: presenter)
             }
+        } else {
+            presentZendesk(from: presenter)
         }
     }
 
+    private func presentZendesk(from presenterViewController: UIViewController) {
+        ZendeskUtils.sharedInstance.showNewRequestIfPossible(
+            from: presenterViewController,
+            with: sourceTag,
+            identityUpdated: { [weak self] identityUpdated in
+                if identityUpdated {
+                    self?.reloadViewModel()
+                }
+            }
+        )
+    }
+
+    private func loadForums() {
+        guard let url = Constants.forumsURL else { return }
```

Comment:
As a side note, I think we could remove the overhead of this `guard` by declaring `forumsURL` as a `URL` and force unwrapping its assignment.

https://github.com/wordpress-mobile/WordPress-iOS/blob/2725b02ab24e48b5102ca7db2194272f650e4286/WordPress/Classes/ViewRelated/Me/Account%20Settings/AccountSettingsViewController.swift#L368

The value is a hardcoded URL and the only way it can be `nil` is if a developer makes an error in spelling out that string. It seems like a good use case for force unwrapping.